noir/noir-repo/compiler/noirc_frontend/src/hir/type_check/mod.rs

```diff
@@ -503,6 +503,7 @@ mod test {
             trait_constraints: Vec::new(),
             direct_generics: Vec::new(),
             is_entry_point: true,
+            should_fold: true,
```

Comment:
```suggestion
            should_fold: false,
```
I think we should keep this as false to better match the previous test behavior